src/common/meta/src/wal_options_allocator/kafka/topic_manager.rs

```diff
@@ -117,7 +131,11 @@ impl TopicManager {
             base: self.config.backoff.base as f64,
             deadline: self.config.backoff.deadline,
         };
-        let client = ClientBuilder::new(self.config.broker_endpoints.clone())
+        let mut broker_endpoints = Vec::with_capacity(self.config.broker_endpoints.len());
+        for endpoint in &self.config.broker_endpoints {
+            broker_endpoints.push(Self::resolve_broker_endpoint(endpoint).await?);
+        }
```

Comment:
Functional programming style is more preferable.

Reply:
The domain resolution should be performed only when the provided broker endpoints are not IPv4 addresses.

Reply:
`lookup_host` https://docs.rs/tokio/latest/src/tokio/net/addr.rs.html#173 will handle this internally.